### ISSUE TITLE
Unable to compile tools on Debian 10

### ISSUE THREAD

**Issue by Zoidwaffle** [azerothcore/azerothcore-wotlk#4527]
##### CURRENT BEHAVIOUR:

I'm unable to compile the tools for the server.

Using the flag `-DTOOLS=1` causes an error for the map extractor.

```
    /usr/bin/ld: CMakeFiles/mapextractor.dir/System.cpp.o: in function `ExtractCameraFiles(int, bool)':
    System.cpp:(.text+0x68b8): undefined reference to `std::filesystem::status(std::filesystem::__cxx11::path const&)'
    /usr/bin/ld: CMakeFiles/mapextractor.dir/System.cpp.o: in function `std::filesystem::__cxx11::path::path<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::filesystem::__cxx11::path>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::filesystem::__cxx11::path::format)':
    System.cpp:(.text._ZNSt10filesystem7__cxx114pathC2INSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEES1_EERKT_NS1_6formatE[_ZNSt10filesystem7__cxx114pathC2INSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEES1_EERKT_NS1_6formatE]+0x8f): undefined reference to `std::filesystem::__cxx11::path::_M_split_cmpts()'
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    make[2]: *** [src/tools/map_extractor/CMakeFiles/mapextractor.dir/build.make:174: src/tools/map_extractor/mapextractor] Error 1
    make[1]: *** [CMakeFiles/Makefile2:1071: src/tools/map_extractor/CMakeFiles/mapextractor.dir/all] Error 2
    make[1]: *** Waiting for unfinished jobs....
    make: *** [Makefile:130: all] Error 2
```

##### EXPECTED BLIZZLIKE BEHAVIOUR:

Compile without errors

##### STEPS TO REPRODUCE THE PROBLEM:

1. Use Debian 10 as server OS
2. Checkout 0a8a7ef1494b1dcc43bdf5134f7c4836acc66f4e (or later) and compile it with `-DTOOLS=1`
3. Cry

##### EXTRA NOTES:

- This used to work up until f07966fd3e208412511c90f16ff2b933a15f8689
- It broke in this commit 0a8a7ef1494b1dcc43bdf5134f7c4836acc66f4e

##### AC HASH/COMMIT:

Just tested, again, on 97b8394f97d8505284ded67c48541230ed12b3a1 but the exact commit that broke it is listed above.

##### OPERATING SYSTEM:

Debian 10.

##### MODULES:

mod-account-achievements
mod-ah-bot
mod-arac
mod-autobalance
mod-congrats-on-level
mod-morph-all-players
mod-morphsummon
mod-npc-buffer
mod-transmog
mod-weapon-visual

**Comment by FrancescoBorzi** [comment]
@Zoidwaffle what's the version of your compiler and cmake?

**Comment by Zoidwaffle** [comment]
`cmake version 3.16.3`
and
```
clang version 7.0.1-8+deb10u2 (tags/RELEASE_701/final)
Target: x86_64-pc-linux-gnu
```

**Comment by FrancescoBorzi** [comment]
@Zoidwaffle can you try with a newer version of clang?

**Comment by Zoidwaffle** [comment]
Yeah, no luck.

`clang version 8.0.1-3~bpo10+1 (tags/RELEASE_801/final)`

Thanks for your suggestions so far. I will try to troubleshoot this more later.

**Comment by iThorgrim-Hub** [comment]
Fix this probleme with this solution.

Create new Source List :
`sudo nano /etc/apt/sources.list.d/testing.list`

Past this in your new Source List : 
`deb http://deb.debian.org/debian testing main`

Paste this in your terminal : 
`sudo apt update && sudo apt upgrade && sudo apt install testing build-essential && sudo apt install clang`

Now compilation work 100% for Debian 10, test work perfectly with bfb6faa

**Comment by Zoidwaffle** [comment]
The default build-essential is 12.6 and clang is 7.0.1-8+deb10u2.

I previously tested out the backports for clang - using 8.0.1-3~bpo10+1.

I really don't like using testing on a server, but hey, at least this is a "fun" server, so I guess exceptions can be made :)

I tried to limit the changes making
`/etc/apt/preferences.d/debian-testing`
with content
```
Package: *
Pin: release a=testing
Pin-Priority: 100
```
not to upgrade the entire system at least... and then

`apt update && apt install build-essential clang -t testing`

The above gives me Clang 11.0.1-2 and build-essential 12.9. But it's not enough to make it work... So I ended up making my Debian 10 a "Debian pre-11" upgrading all packages to `testing`... at least it's not `unstable` :(

So it works but it's no longer Debian 10.

I guess I can have a bleeding edge server (but exactly what package that is an issue, I don't know, but it works in Ubuntu 18.04 so why the heck not Debian 10).

@iThorgrim-Hub - Thanks for your suggestion - I'm closing this issue!
